src/models/device.ts

```diff
@@ -368,7 +382,10 @@ const getDeviceModel = function (
 			{ primitive: true, promise: true },
 		);
 
-		const osUpdateHelper = await getOsUpdateHelper();
+		const osUpdateHelper = options.runDetached === true
+		? await getOsUpdateHelperv2()
```

Comment:
I don't see the value of a second getOsUpdateHelper function. Just parameterize getOsUpdateHelper() to include a version number parameter. Use that parameter to pass the right endpoint version to DeviceActionsService. Also, for version 2, don't return a getOsUpdateStatus() function.